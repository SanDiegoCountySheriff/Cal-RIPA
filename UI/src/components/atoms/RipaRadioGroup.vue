<template>
  <v-radio-group v-model="model" hide-details>
    <v-radio
      v-for="(item, index) in items"
      :key="index"
      :label="item.name"
      :value="item.value"
    ></v-radio>
  </v-radio-group>
</template>

<script>
export default {
  name: 'ripa-radio-group',

  data() {
    return {
      viewModel: this.value,
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newVal) {
        this.viewModel = newVal
        this.$emit('input', newVal)
      },
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = newVal
    },
  },

  props: {
    value: {
      type: [String, Number],
      default: '',
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
}
</script>
