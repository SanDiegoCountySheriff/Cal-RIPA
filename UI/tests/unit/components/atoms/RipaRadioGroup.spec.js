import RipaRadioGroup from '@/components/atoms/RipaRadioGroup'
import Vuetify from 'vuetify'
import { shallowMount, mount } from '@vue/test-utils'

describe('Ripa Radio Group', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })

  const factory = propsData => {
    return shallowMount(RipaRadioGroup, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaRadioGroup, { vuetify })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set model', async () => {
    wrapper = factory()
    wrapper.vm.model = 'New Value'

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual('New Value')
    expect(wrapper.emitted('input')).toBeTruthy()
  })

  it('should handle clear selection', () => {
    wrapper = factory()
    wrapper.vm.handleClearSelection()

    expect(wrapper.vm.model).toEqual(null)
  })

  it('should watch value', async () => {
    wrapper = factory()

    wrapper.vm.value = 'New Value'
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual('New Value')
  })
})
