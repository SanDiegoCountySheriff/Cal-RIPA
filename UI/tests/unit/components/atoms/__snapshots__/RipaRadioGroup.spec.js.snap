// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Radio Group should match snapshot 1`] = `
<div class="ripa-radio-group">
  <v-radio-group-stub valuecomparator="[Function]" value="" activeclass="v-item--active" tag="div" errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" height="auto" column="true"></v-radio-group-stub>
  <!---->
</div>
`;
